src/main/java/com/bookclub/util/DatabaseManager.java

```
@@ -5,36 +5,42 @@
 import java.sql.SQLException;
 
 public class DatabaseManager {
-    // Singleton implementation.
     private static DatabaseManager instance = null;
     private Connection connection;
-    private static final String URL = "jdbc:sqlite:bookclub.db";
-    
+    private static final String URL = "jdbc:sqlite:book_club_app.db"; // Path to your SQLite DB
+
     private DatabaseManager() {
         try {
             connection = DriverManager.getConnection(URL);
-        } catch (SQLException sqlEx) {
-            System.err.println(sqlEx);
+        } catch (SQLException e) {
+            System.err.println("Failed to establish connection: " + e.getMessage());
         }
     }
-    
+
     public static DatabaseManager getInstance() {
         if (instance == null) {
-            instance = new DatabaseManager(); // I personally hate this approach.
+            instance = new DatabaseManager();
         }
         return instance;
     }
 
     public Connection getConnection() {
+        try {
+            if (connection == null || connection.isClosed()) {
```

This shouldn't ever be hit and tbh should throw an error instead of silently handling. If we have no initialisation in a constructed method somethings gone wrong.